contracts/strategies/StrategyMKRVaultDAIDelegate.sol

```diff
@@ -286,35 +311,30 @@ contract StrategyMKRVaultDAIDelegate {
     }
 
     function harvest() public {
-        require(msg.sender == strategist || msg.sender == harvester || msg.sender == governance, "!authorized");
+        require(msg.sender == strategist || msg.sender == keeper || msg.sender == governance, "!authorized");
 
         uint256 v = getUnderlyingDai();
         uint256 d = getTotalDebtAmount();
         require(v > d, "profit is not realized yet!");
         uint256 profit = v.sub(d);
 
-        uint256 _before = IERC20(want).balanceOf(address(this));
-        _swap(_withdrawDaiMost(profit));
-        uint256 _after = IERC20(want).balanceOf(address(this));
+        _withdrawDaiMost(profit);
+        _swap(IERC20(dai).balanceOf(address(this)));
 
-        uint256 _want = _after.sub(_before);
+        uint256 _want = IERC20(want).balanceOf(address(this));
         if (_want > 0) {
-            uint256 _fee = _want.mul(performanceFee).div(performanceMax);
-            uint256 _strategistReward = _fee.mul(strategistReward).div(strategistRewardMax);
+            uint256 _fee = _want.mul(performanceFee).div(c_base);
+            uint256 _strategistReward = _want.mul(strategistReward).div(c_base);
             IERC20(want).safeTransfer(strategist, _strategistReward);
-            IERC20(want).safeTransfer(IController(controller).rewards(), _fee.sub(_strategistReward));
+            IERC20(want).safeTransfer(IController(controller).rewards(), _fee);
         }
 
         deposit();
     }
 
     function shouldDraw() external view returns (bool) {
-        uint256 _safe = c.mul(1e2);
-        uint256 _current = getmVaultRatio(0);
-        if (_current > c_base.mul(c_safe).mul(1e2)) {
-            _current = c_base.mul(c_safe).mul(1e2);
-        }
-        return (_current > _safe);
+        // 5% buffer to avoid deposit/rebalance loops
+        return (getmVaultRatio(0) > c.mul(1e2).mul(c_base).div((c_base.sub(buffer))));
```

Comment:
What is the implication of this check being equal? since i see `shouldRepay` does the same calc only is a `lt` operation.